src/main/java/com/zufar/onlinestore/OnlineStoreApplication.java

```diff
@@ -1,14 +1,22 @@
 package com.zufar.onlinestore;
 
 import com.zufar.onlinestore.payment.config.StripeConfiguration;
+import com.zufar.onlinestore.product.repository.ProductInfoRepository;
+import com.zufar.onlinestore.reservation.config.ReservationTimeoutConfiguration;
+import com.zufar.onlinestore.reservation.service.ReservationCreator;
+import lombok.AllArgsConstructor;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 
 @SpringBootApplication
-@EnableConfigurationProperties(StripeConfiguration.class)
+@AllArgsConstructor
+@EnableConfigurationProperties(value = {StripeConfiguration.class, ReservationTimeoutConfiguration.class})
 public class OnlineStoreApplication {
 
+    private final ReservationCreator reservationCreator;
```

Comment:
Is this improvement part of the task? I will redo everything - answer below.

Reply:
You can reove this field from OnlineStoreApplication class I just tested code and forgot to remove this field.

But,

We exctract all service methods into another class
ReservationCreator - is responsible to create reservation
Also you need:
ReservationCanceller - is responsible to cancelreservation
ReservationConfirmer - is responsible to confirm reservation